### PR TITLE
Adding table of techniques to readme

### PR DESCRIPTION
**Pull request by crazy4pi314**
Description
-----------
Adds the table as per group meeting discussion.

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

### CODE REVIEW

**Review by rmlarose — COMMENTED** [review]
Thanks Sarah! On https://github.com/unitaryfund/mitiq/tree/docs/technique-table I see some CSS in text:

image

If the table includes planned but not completed techniques (like dynamical decoupling), might be good to update the intro text (i.e, update "Mitiq currently implements"). 

Optional: Maybe "Date" -> "Date completed" or something similar. "Date completed" might be too long though.

**Comment by crazy4pi314** [comment]
@rmlarose some quick comments:
> Thanks Sarah! On https://github.com/unitaryfund/mitiq/tree/docs/technique-table I see some CSS in text:
> 
> image
> 
If we want to keep the coloring from the gdoc before we need css. I can either move to the end, or we can drop the coloring.

> If the table includes planned but not completed techniques (like dynamical decoupling), might be good to update the intro text (i.e, update "Mitiq currently implements").
> 
Done
> Optional: Maybe "Date" -> "Date completed" or something similar. "Date completed" might be too long though.

Done

**Comment by rmlarose** [comment]
> If we want to keep the coloring from the gdoc before we need css. I can either move to the end, or we can drop the coloring.

I don't [see](https://github.com/unitaryfund/mitiq/tree/docs/technique-table#error-mitigation-techniques) the coloring even with the CSS:

image

So I think it's fine to drop the CSS. Otherwise LGTM. Thanks for adding the docs column, this is great to show.

**Comment by crazy4pi314** [comment]
Weird.... cool, I'll drop it then. Much better for accessibility anyway!

**Review by rmlarose — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -123,13 +123,17 @@ Error in simulation (with mitigation): 0.000519
...
+|-------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------|----------------|
+| Zero-noise extrapolation                  | [ZNE](https://mitiq.readthedocs.io/en/stable/guide/guide-zne.html)                                                                    | [mitiq.zne](https://github.com/unitaryfund/mitiq/tree/master/mitiq/zne) | [1612.02058](https://arxiv.org/abs/1612.02058)                                                      | July 2020      |
+| Probabilistic error cancellation          | [PEC](https://mitiq.readthedocs.io/en/stable/guide/guide-getting-started.html#error-mitigation-with-probabilistic-error-cancellation) | [mitiq.pec](https://github.com/unitaryfund/mitiq/tree/master/mitiq/pec) | [1612.02058](https://arxiv.org/abs/1612.02058)                                                      | Jan 2021       |
```
Comment (rmlarose) [inline README.md:131]:
```suggestion
| Probabilistic error cancellation          | [PEC](https://mitiq.readthedocs.io/en/stable/guide/guide-getting-started.html#probabilistic-error-cancellation) | [mitiq.pec](https://github.com/unitaryfund/mitiq/tree/master/mitiq/pec) | [1612.02058](https://arxiv.org/abs/1612.02058)                                                      | Jan 2021       |
```

**Pull request merged**